### PR TITLE
Misc bitsets

### PR DESCRIPTION
**Pull request by chrisforbes**
More HashSet<string> -> BitSet<T> conversions

### CODE REVIEW

**Comment by chrisforbes** [comment]
Bunch of churn here; apparently I like to drop random semicolons after class definitions...

**Review by GraionDilach — DISMISSED** [review]
:+1:

**Comment by pchote** [comment]
Is it just me, or does this break cloak detection in the RA mod? (test with phase transports vs radar domes)

**Comment by teinarss** [comment]
The stealth detection works now in TD.

File: OpenRA.Mods.Cnc/Traits/Buildings/ClonesProducedUnits.cs [hunk 1]
```diff
@@ -22,7 +21,7 @@ public class ClonesProducedUnitsInfo : ITraitInfo, Requires<ProductionInfo>, Req
 	{
 		[FieldLoader.Require]
 		[Desc("Uses the \"Cloneable\" trait to determine whether or not we should clone a produced unit.")]
-		public readonly HashSet<string> CloneableTypes = new HashSet<string>();
+		public readonly BitSet<CloneableType> CloneableTypes = default(BitSet<CloneableType>);
```
Comment (GraionDilach) [inline OpenRA.Mods.Cnc/Traits/Buildings/ClonesProducedUnits.cs]: Isn't this missing a ( ?

**Comment by GraionDilach** [comment]
Nvm me today, I'm blind.

**Review by GraionDilach — APPROVED** [review]
:+1: again

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/0a7d76f9d2dcd96a6ad1bb3f5d4eb913527228aa...59c65a2c274a673644cf593359595a7ff1e364f0)

**Pull request merged**